app/src/main/java/com/example/fuji/RegisterFragment.kt

```diff
@@ -27,12 +29,58 @@ class RegisterFragment : Fragment() {
 
-        view.findViewById<ImageView>(R.id.register_button_background).setOnClickListener {
-            findNavController().navigate(R.id.action_RegisterFragment_to_LoginFragment)
-        }
+        val emailPattern = "[a-zA-Z0-9._-]+@[a-z]+\\.+[a-z]+".toRegex()
+        val emailInput = view.findViewById<EditText>(R.id.register_email_entry).text
+        val firstNameInput = view.findViewById<EditText>(R.id.register_first_name_entry).text
+        val lastNameInput = view.findViewById<EditText>(R.id.register_last_name_entry).text
+        val passwordInput = view.findViewById<EditText>(R.id.register_password_entry).text
+        val confirmPasswordInput = view.findViewById<EditText>(R.id.register_confirm_password_entry).text
+
+        view.findViewById<Button>(R.id.register_button).setOnClickListener {
 
-        view.findViewById<TextView>(R.id.register_button).setOnClickListener() {
-            findNavController().navigate(R.id.action_RegisterFragment_to_LoginFragment)
+            if(firstNameInput.toString().trim().isEmpty()) {
+                Toast.makeText(context, "Enter first name", Toast.LENGTH_SHORT).show()
+            } else {
+                if(lastNameInput.toString().trim().isEmpty()) {
+                    Toast.makeText(context, "Enter last name", Toast.LENGTH_SHORT).show()
+                } else {
+                    if(passwordInput.toString().trim().isEmpty()) {
+                        Toast.makeText(context, "Enter a password", Toast.LENGTH_SHORT).show()
+                    } else {
+                        if(confirmPasswordInput.toString().trim().isEmpty()) {
+                            Toast.makeText(context, "Enter a password confirmation", Toast.LENGTH_SHORT).show()
+                        } else {
+                            if(passwordInput.toString().trim() != confirmPasswordInput.toString().trim()) {
+                                Toast.makeText(context, "Confirmation does not match password", Toast.LENGTH_SHORT).show()
+                            } else {
+                                if(emailInput.toString().trim().isEmpty()) {
+                                    Toast.makeText(context, "Enter email address", Toast.LENGTH_SHORT).show()
+                                } else {
+                                    if(emailInput.toString().matches(emailPattern)) {
+                                        Toast.makeText(context, "Valid email address", Toast.LENGTH_SHORT).show()
+                                        Handler().postDelayed({findNavController().navigate(R.id.action_RegisterFragment_to_LoginFragment)}, 2000)
+                                    } else {
+                                        Toast.makeText(context, "Invalid email address", Toast.LENGTH_SHORT).show()
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
```

Comment:
so a cleaner way of doing this would be 
```suggestion
            if(firstNameInput.toString().trim().isEmpty()) {
                Toast.makeText(context, "Enter first name", Toast.LENGTH_SHORT).show()
                return;
            } 
            
             if(lastNameInput.toString().trim().isEmpty()) {
               Toast.makeText(context, "Enter last name", Toast.LENGTH_SHORT).show()
               return;
             }
             
             if(passwordInput.toString().trim().isEmpty()) {
               Toast.makeText(context, "Enter a password", Toast.LENGTH_SHORT).show()
               return;
              }
              
              if(confirmPasswordInput.toString().trim().isEmpty()) {
                Toast.makeText(context, "Enter a password confirmation", Toast.LENGTH_SHORT).show()
                return;
               }
                
               if(passwordInput.toString().trim() != confirmPasswordInput.toString().trim()) {
                 Toast.makeText(context, "Confirmation does not match password", Toast.LENGTH_SHORT).show()
                 return;
                }
                
                if(emailInput.toString().trim().isEmpty()) {
                  Toast.makeText(context, "Enter email address", Toast.LENGTH_SHORT).show()
                  return;
                 }
                 
                 if(!emailInput.toString().matches(emailPattern)) {
                   Toast.makeText(context, "Invalid email address", Toast.LENGTH_SHORT).show()
                    return;
                  } 
                  
                  
                   Toast.makeText(context, "Valid email address", Toast.LENGTH_SHORT).show()
                    Handler().postDelayed({findNavController().navigate(R.id.action_RegisterFragment_to_LoginFragment)}, 2000)
```

Reply:
I believe it should be fixed now